packages/logs/src/domain/trackNetworkError.ts

```diff
@@ -46,6 +50,38 @@ export function trackNetworkError(configuration: LogsConfiguration, errorObserva
   }
 }
 
+// TODO: ideally, computeResponseData should always call the `callback` with a string instead of
+// `any`. But to keep retrocompatibility, in the case of XHR with a `responseType` different than
+// "text", the response data should be whatever `xhr.response` is. This is a bit confusing as Logs
+// event 'stack' is expected to be a string. This should be changed in a future major version as it
+// could be a breaking change.
+export function computeResponseData(
+  { xhr, response, error }: { xhr?: XMLHttpRequest; response?: Response; error?: unknown },
+  configuration: LogsConfiguration,
+  callback: (responseData?: any) => void
+) {
+  if (xhr) {
+    if (typeof xhr.response === 'string') {
+      callback(truncateResponseText(xhr.response, configuration))
+    } else {
+      callback(xhr.response)
+    }
+  } else if (response) {
+    response
+      .clone()
+      .text()
+      .then(
+        (text) => callback(text),
+        (error) => callback(`Unable to retrieve response: ${error as string}`)
+      )
+  } else if (error) {
+    callback(truncateResponseText(toStackTraceString(computeStackTrace(error)), configuration))
+  } else {
+    // This case should not happen in practice, but better safe than sorry.
+    callback()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Could it be related to the 'Failed to load' case? or is it something else?
https://github.com/DataDog/browser-sdk/blob/23b7b2779c44ce1a157d3f1cdcf3f4b5401bc631/packages/logs/src/domain/trackNetworkError.ts#L38

Reply:
It can't happen because `XhrCompleteContext | FetchCompleteContext` has always either response, error or xhr defined, but those types alone are not enough to guarantee it. Maybe we should improve typings and have something like:

```ts
interface FetchErrorContext {
  error: Error
...
}

interface FetchCompleteContext {
  response: Response
...
}
```

and then using `XhrCompleteContext | FetchErrorContext | FetchCompleteContext`, so TS can ensure than at least one of the three is define?

Reply:
hum ok, so 'Failed to load' would only be used when responseData is `''`, we should not have undefined value.
Then, yes, it could be nice to avoid this empty callback if it is not needed or add a monitoring log if we have some doubts.

Reply:
Ok to remove the empty callback. The test case still make sense, because as you observed an empty response still needs the "Failed to load" fallback, right?

Reply:
Yes, but `responseText: undefined` should then be `responseText: ''` and we could clarify the test description.
it could worth a quick chat if I am missing something.